lib/rsa.js

```diff
@@ -887,6 +889,30 @@ pki.rsa.generateKeyPair = function(bits, e, options, callback) {
   // key generation code if available and if parameters are acceptable
   if(!forge.options.usePureJavaScript && callback &&
     bits >= 256 && bits <= 16384 && (e === 0x10001 || e === 3)) {
+    if(forge.util.isNodejs && _detectNodeCrypto('generateKeyPair')) {
+      return _crypto.generateKeyPair('rsa', {
+        modulusLength: bits,
+        publicExponent: e,
+        publicKeyEncoding: {
+          type: 'pkcs1',
+          format: 'der'
+        },
+        privateKeyEncoding: {
+          type: 'pkcs8',
+          format: 'der'
+        }
+      }, function(err, pub, pkcs8) {
+        if (err) {
+          return callback(err);
+        }
+        var privateKey = pki.privateKeyFromAsn1(
```

Comment:
Is there a shorter route from whatever format `generateKeyPair` might generate to a `privateKey`?

Reply:
Looks like I can ask Node for a `pem` and then use `pki.privateKeyFromPem()`.